### PR TITLE
feat(List): selectedIndex && a11y enhancements

### PR DESCRIPTION
**Pull request by jensborch**
This should correspond to the improvements in https://github.com/rmwc/rmwc/pull/836

### CODE REVIEW

**Comment by jensborch** [comment]
@EmiBemi @b-d-m-p This should be close to ready so I would like help testing

File: packages/list/src/lib/foundation.tsx [hunk 1]
```diff
@@ -101,23 +108,42 @@ export const useListFoundation = (props: ListProps & React.HTMLProps<any>) => {
...
           }
         },
         addClassForElementIndex: (index: number, className: string) => {
-          const element = listElements(rootEl.ref)[index];
-          if (element) {
-            element.classList.add(deprecatedClassNameMap[className]);
-          }
+          setListItemClasses((listItems: ListItemClassesState) => {
+            if (listItems[index] && listItems[index].indexOf(className) > -1) {
+              // no dupes
```
Comment (EmiBemi) [inline packages/list/src/lib/foundation.tsx:119]: This comment is probably not needed.

File: packages/list/src/lib/foundation.tsx [hunk 2]
```diff
@@ -292,10 +318,51 @@ export const useListFoundation = (props: ListProps & React.HTMLProps<any>) => {
   rootEl.setProp('onFocus', handleFocusIn, true);
   rootEl.setProp('onBlur', handleFocusOut, true);
 
+  const initializeListRole = useCallback((): void => {
```
Comment (EmiBemi) [inline packages/list/src/lib/foundation.tsx:321]: Is type `: void` needed? I would assume it's inferred.

File: packages/list/src/lib/foundation.tsx [hunk 3]
```diff
@@ -292,10 +318,51 @@ export const useListFoundation = (props: ListProps & React.HTMLProps<any>) => {
...
+    }
+
+    const hasSelectedIndex_: boolean =
+      !!props.selectedIndex ||
+      !!rootEl.ref.querySelector(`
+        .${MDCListFoundation.cssClasses.LIST_ITEM_ACTIVATED_CLASS},
+        .${MDCListFoundation.cssClasses.LIST_ITEM_SELECTED_CLASS}
+      `);
+
+    const isCheckboxList_: boolean = !!rootEl.ref.querySelector(
```
Comment (EmiBemi) [inline packages/list/src/lib/foundation.tsx:342]: Same here. Type is not needed. Same as above and below.

File: packages/list/src/lib/list.tsx [hunk 4]
```diff
@@ -56,12 +67,39 @@ export const List = createComponent<ListProps>(function List(props, ref) {
...
   } = props;
-  const { rootEl } = useListFoundation({
-    ...props,
-    wrapFocus
-  });
+  const { rootEl, listItemClasses, setEnabled, role } =
+    useListFoundation(props);
+
+  const getListItemRole = (): { role?: string } => {
+    // TODO(mgr34): menuitems with checkboxs or radios should be
```
Comment (EmiBemi) [inline packages/list/src/lib/list.tsx:79]: Should we move this to an issue?

**Review by EmiBemi — APPROVED** [review]

**Review by EmiBemi — APPROVED** [review]

**Review by EmiBemi — APPROVED** [review]

**Review by EmiBemi — DISMISSED** [review]

**Pull request closed without merging**